RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -104,6 +108,11 @@ PixelCPEClusterRepair::PixelCPEClusterRepair(edm::ParameterSet const & conf,
      recommend2D_.push_back(str);
    }
 
+   // do not recommend 2D if theMagField!=3.8
+   if(theMagField>=3.65 && theMagField<3.9){
+     recommend2D_.clear();
+   }
```

Comment:
Hi @OzAmram @cmantill, maybe I misread the code, but should not the logic of this clause be reversed?
Right now you are clearing `recommend2D_` if the magnetic field **is** the nominal one.

Reply:
@mmusich you are right, sorry for the bug. It was introduced when I was cleaning up the code after testing and did an incorrect translation of the logic.. https://github.com/cmantill/cmssw/commit/c78b14819bdfac71bdc324b062d160b13845a962#diff-34f19ce9486ae317e412a733351d2274R112. I pushed a fix and Oz will pick it up asap.